docs/hub-resolver.md

```diff
@@ -126,6 +126,36 @@ spec:
   # overall will not succeed without those parameters.
 ```
 
+### Version constraint
+
+Instead of a version you can specify a constraint to choose from. The constraint is a string as documented in the [go-version](https://github.com/hashicorp/go-version) library.
+
+Some examples:
+
+```yaml
+params:
+  - name: name
+    value: git-clone
+  - name: version
+    value: ">=0.7.0"
+```
+
+Will only choose the git-clone task that is greater than version `0.7.0`
+
+```yaml
+params:
+  - name: name
+    value: git-clone
+  - name: version
+    value: ">=0.7.0, < 2.0.0"
+```
+
+Will select the git-clone task that is greater than version `0.7.0` and less than version `2.0.0`
```

Comment:
currently, we have up to 0.9.0 -- so three versions meet this constraint -- which version will be chosen? read through https://github.com/hashicorp/go-version but it wasn't clear to me

Reply:
ah yeah i meant to add that to the documentation but forgot, the code has some logic to only choose the latest one as tested here https://github.com/tektoncd/pipeline/pull/7257/files#diff-8867ae56005554e8425016bb15b437a7b2b17b9d763572ce20f0dc9c974ec297R247

will amend the doc,